kernel/realsense/d4xx.c

```diff
@@ -4824,18 +4824,31 @@ static ssize_t ds5_dfu_device_read(struct file *flip,
 {
 	struct ds5 *state = flip->private_data;
 	u16 fw_ver, fw_build;
-	char msg[32];
+	char msg[64];
 	int ret = 0;
+	struct __fw_status f = {0};
 
 	if (mutex_lock_interruptible(&state->lock))
 		return -ERESTARTSYS;
-	ret |= ds5_read(state, DS5_FW_VERSION, &fw_ver);
-	ret |= ds5_read(state, DS5_FW_BUILD, &fw_build);
-	if (ret < 0)
-		goto e_dfu_read_failed;
-	snprintf(msg, sizeof(msg) ,"DFU info: \tver:  %d.%d.%d.%d\n",
+	if (state->dfu_dev.dfu_state_flag == DS5_DFU_RECOVERY) {
+		ds5_write_with_check(state, 0x500c, 0x00);
```

Comment:
1. What does 0x500c means?
Maybe we should keep it as a define with a proper name?
2. Are we entering DFU state here? what didn't happened before?

Reply:
1. This used in ds5_dfu_detach L4803, in order to get properly unlocked registers to read  ds5_dfu_get_dev_info.
2. We verify if camera in DFU state we can read DFU version. If camera in operation mode we cannot enter DFU and restore to operational. There is no opcode to exit DFU.